src/thing.test.ts

```diff
@@ -958,18 +968,40 @@ describe("asIri", () => {
   });
 
   it("returns the IRI of a local Thing relative to a given base IRI", () => {
-    const localThing = Object.assign(dataset(), { name: "some-name" });
+    const localSubject: LocalNode = Object.assign(DataFactory.blankNode(), {
+      name: "some-name",
+    });
+    const localThing = Object.assign(dataset(), { localSubject: localSubject });
 
     expect(asUrl(localThing, "https://some.pod/resource")).toBe(
       "https://some.pod/resource#some-name"
     );
   });
 
   it("throws an error when a local Thing was given without a base IRI", () => {
-    const localThing = Object.assign(dataset(), { name: "some-name" });
+    const localSubject: LocalNode = Object.assign(DataFactory.blankNode(), {
+      name: "some-name",
+    });
+    const localThing = Object.assign(dataset(), { localSubject: localSubject });
 
     expect(() => asUrl(localThing, undefined as any)).toThrow(
       "The URL of a Thing that has not been persisted cannot be determined without a base URL."
     );
   });
 });
+
+describe("toNode", () => {
+  it("should result in equal LocalNodes for the same ThingLocal", () => {
+    const localSubject: LocalNode = Object.assign(
+      DataFactory.blankNode("Arbitrary blank node"),
+      { name: "localSubject" }
+    );
+    const thing: ThingLocal = Object.assign(dataset(), {
+      localSubject: localSubject,
+    });
+    const node1 = toNode(thing);
+    const node2 = toNode(thing);
+    expect(node1.equals(node2)).toBe(true);
+    expect(node1).toEqual(node2);
```

Comment:
Is there a difference between these two lines ?

Reply:
Yep, the former uses the RDF/JS [`equals`](https://rdf.js.org/data-model-spec/#dom-term-equals) method, which checks that the Nodes are equivalent (i.e. have the same value and Node type), whereas the latter checks that they're actually the same instance.

(The latter does actually not need to be true... I'll remove it.)

Reply:
Fixed in 6c4411e942711039716f8c6442908778ba8e9496.